drivers/sensor/adxl345/adxl345.h

```diff
@@ -49,8 +60,23 @@ struct adxl345_sample {
 	int16_t z;
 };
 
-struct adxl345_dev_config {
+union adxl345_bus {
```

Comment:
I'm just having a general question about style and consistency here: Should we have these named unions for dual bus support or not? I've been thinking about removing these while doing cleanup since I've felt that `i2c_write_dt(&config->i2c,...` is a bit cleaner than `i2c_write_dt(&config->bus.i2c,...`.

Reply:
Hi, @benjaminbjornsson ,

> I'm just having a general question about style and consistency here: Should we have these named unions for dual bus support or not? I've been thinking about removing these while doing cleanup since I've felt that `i2c_write_dt(&config->i2c,...` is a bit cleaner than `i2c_write_dt(&config->bus.i2c,...`.

It intending to reduce memory size.
The sensor work on either the SPI bus or I2C bus. So no need to keep both i2c_dt_spec and spi_dt_spec.
Using union requires less memory than creating both the i2c_dt_spec and spi_dt_spec fields.
(From another point of view, it is an representation that has either SPI or I2C context.)

This style is refered to [bme280](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/bme280/bme280.h#L23)
It is a bit verbose, but the correct style, I think.
(I think it's rare, but it is effective well when both the I2C and SPI buses are enabled.)

Reply:
I agree that the union should be there. I was just thinking about whether we want to use anonymous unions or not.
Could as well use:
`union {
        struct i2c_dt_spec i2c;
        struct spi_dt_spec spi;
};`
and still save that space. Other drivers use anonymous unions and that's why I asked cause I think we should be using one of them and not both for consistency. However, It's not a question that should stop this PR.

Reply:
@benjaminbjornsson ,
Thank you for your comment.

Since there is a function that takes this 'union' as an argument, Let me leave it as it is.